daml-lf/testing-tools/src/main/scala/com/digitalasset/daml/lf/engine/testing/SemanticTester.scala

```diff
@@ -195,34 +194,63 @@ class SemanticTester(
                       "exercise event",
                       scenarioNode,
                       state.remainingLedgerEventIds)
-                  // create synthetic exercise event, again rewriting the appropriate bits. note that we intentionally
-                  // blank the children because we compare them in the recursive call anyway.
-                  val scenarioExerciseEvent = ExerciseEvent(
-                    state.scenarioCoidToLedgerCoid(scenarioExercisesNode.targetCoid),
-                    scenarioExercisesNode.templateId,
-                    scenarioExercisesNode.choiceId,
-                    scenarioExercisesNode.chosenValue.mapContractId(state.scenarioCoidToLedgerCoid),
-                    scenarioExercisesNode.actingParties,
-                    scenarioExercisesNode.consuming,
-                    ImmArray.empty,
-                    scenarioExercisesNode.stakeholders intersect scenarioWitnesses(scenarioNodeId),
-                    scenarioWitnesses(scenarioNodeId),
-                  )
-                  val ledgerExerciseEventToCompare =
-                    ledgerExerciseEvent.copy(children = ImmArray.empty, stakeholders = Set.empty)
-                  val comparedScenarioExerciseEvent =
-                    scenarioExerciseEvent.copy(stakeholders = Set.empty)
-                  if (comparedScenarioExerciseEvent != ledgerExerciseEventToCompare) {
-                    throw SemanticTesterError(
-                      reference,
-                      s"Expected exercise event $comparedScenarioExerciseEvent but got $ledgerExerciseEventToCompare"
-                    )
+
+                  // Check if the contract ID of the result of exercising the contract choice can be mapped.
+                  // If the contract ID cannot be mapped, then delay the check until the children have been processed
+                  // which will add the Contract ID mappings required to test the exercise result.
+                  // Note: If the exercise choice is not defined, then we cannot determine if the children nodes/events
+                  // have been added or not. Perhaps we could keep track of nodes/events that have been processed to
+                  // avoid this.
+                  scenarioExercisesNode.exerciseResult
```

Comment:
this is wrong, since it's not the case that the exercise return value _always_ contains a child contract id

Reply:
Done